lib/facter/self_service.rb

```diff
@@ -188,4 +188,19 @@
     # Is puppet_metrics_collector::system configured
     { S0040: PuppetSelfService.service_running_enabled('puppet_system_processes-metrics.timer') }
   end
+
+  chunk(:S0027) do
+    next unless PuppetSelfService.primary?
+    # Check if thundering herd is occuring.
+    thundering_query_output = PuppetSelfService.psql_thundering_herd.split(' ').drop(2).map(&:to_i)
+    thundering_query_output.delete(0)
+    thundering_occured = false
+    minvalue = thundering_query_output.min.to_i
+    maxvalue = thundering_query_output.max.to_i
+    differences = ((maxvalue - minvalue) * 100) / minvalue
```

Comment:
when minvalue = 0 this crashes:

` Error while resolving custom fact fact='self_service', resolution='<anonymous>': divided by 0`

You need to account for systems with low activity

Reply:
I did handle this minvalue=0 case, The statement below `thundering_query_output.delete(0)` helps to remove the 0 from the output data.

```
    thundering_query_output.delete(0)
    thundering_occured = false
    minvalue = thundering_query_output.min.to_i
```

Reply:
does this confine statement correct? `next unless PuppetSelfService.primary?` i see from one of my agent nodes that the facter is executed and getting `Error: Facter: error while resolving custom facts in /opt/puppetlabs/puppet/cache/lib/facter/self_service.rb: divided by 0`


```
chunk(:S0027) do
    next unless PuppetSelfService.primary?
    # Check if thundering herd is occuring.
    thundering_query_output = PuppetSelfService.psql_thundering_herd.split(' ').drop(2).map(&:to_i)
    thundering_query_output.delete(0)
    thundering_occured = false
    minvalue = thundering_query_output.min.to_i
    maxvalue = thundering_query_output.max.to_i
    differences = ((maxvalue - minvalue) * 100) / minvalue
    if differences >= 75
      thundering_occured = true
    end
    { S0027: thundering_occured }
  end
```

Reply:
@asselvakumar this strange, as the whole fact is confined to just infrastructure nodes. 

   next unless PuppetSelfService.primary? is a further refinement per chunk, but the Whole fact is confined by:

 ` confine { Facter.value(:pe_build) }` which is never true on an normal agent

Reply:
> I did handle this minvalue=0 case, The statement below `thundering_query_output.delete(0)` helps to remove the 0 from the output data.
> 
> ```
>     thundering_query_output.delete(0)
>     thundering_occured = false
>     minvalue = thundering_query_output.min.to_i
> ```

The acceptance tests are still failing for a divide by zero error, which implies there are still 0's leaking into the function